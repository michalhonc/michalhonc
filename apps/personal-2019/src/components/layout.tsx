import * as React from "react";

import { Footer } from "./footer";
import "../styles/main.scss";

export const Layout: React.FC = ({ children }) => (
  <div className="Layout">
    <div>
      <main>{children}</main>
      <Footer />
    </div>
  </div>
);
