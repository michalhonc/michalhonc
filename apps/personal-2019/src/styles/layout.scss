*::selection {
  background: var(--c-alpha-light);
}

.Layout {
  overflow: hidden;
}

.Container {
  max-width: 102.4rem;
  margin: 0 auto;
  padding-top: 12rem;

  @include respond('tab-port') {
    padding-top: 6rem;
  }
}

.underline {
  box-shadow: inset 0 -0.6em var(--c-alpha-light);
}

h2 {
  font-size: 3.6rem;
  color: var(--c-beta);
  text-align: center;
  font-weight: bold;

  @include respond('phone') {
    font-size: 2rem;
    padding: 0 1rem;
  }
}

h3 {
  font-size: 2.6rem;
  color: var(--c-beta);
  font-weight: bold;

  @include respond('phone') {
    font-size: 1.8rem;
  }
}
