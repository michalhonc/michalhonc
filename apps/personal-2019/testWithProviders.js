import React from "react";
import renderer from "react-test-renderer";
import { IntlContextProvider } from "gatsby-plugin-intl";

export const renderWithProviders = (component) =>
  renderer.create(
    <IntlContextProvider value={{ language: "en-US", routed: true }}>
      {component}
    </IntlContextProvider>
  );
