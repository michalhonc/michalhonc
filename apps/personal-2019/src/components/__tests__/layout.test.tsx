import * as React from "react";
import renderer from "react-test-renderer";
import { renderWithProviders } from "../../../testWithProviders";
import { Layout } from "../layout";

describe("layout", () => {
  it("renders correctly", () => {
    const tree = renderWithProviders(<Layout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
