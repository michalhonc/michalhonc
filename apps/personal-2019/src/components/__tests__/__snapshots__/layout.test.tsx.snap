// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout renders correctly 1`] = `
<div
  className="Layout"
>
  <div>
    <main />
    <footer
      id="kontakty"
    >
      <test-file-stub />
      <ul>
        <li>
          <a
            aria-label="mobile"
            href="tel:+420733209581"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            aria-label="WhatsApp"
            href="https://wa.me/420733209581"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            aria-label="Twitter"
            href="https://twitter.com/michalhonc"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            aria-label="LinkedIn"
            href="https://linkedin.com/in/michalhonc/"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            aria-label="Instagram"
            href="https://instagram.com/michalhonc.cz/"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            aria-label="E-mail"
            href="mailto:info@michalhonc.cz"
          >
            <test-file-stub />
          </a>
        </li>
      </ul>
      <small
        className="underline"
      >
        2020
         
        Michal Honc |
         
        <a
          href="mailto:info@michalhonc.cz"
          title="E-mail"
        >
          info@michalhonc.cz
        </a>
      </small>
    </footer>
  </div>
</div>
`;
